demo/run-osm-multicluster-demo.sh

```diff
@@ -0,0 +1,174 @@
+DEPLOY_TRAFFIC_SPLIT="${DEPLOY_TRAFFIC_SPLIT:-true}"
+CTR_TAG="${CTR_TAG:-$(git rev-parse HEAD)}"
+IMAGE_PULL_POLICY="${IMAGE_PULL_POLICY:-Always}"
+ENABLE_DEBUG_SERVER="${ENABLE_DEBUG_SERVER:-true}"
+ENABLE_EGRESS="${ENABLE_EGRESS:-false}"
+DEPLOY_GRAFANA="${DEPLOY_GRAFANA:-false}"
+DEPLOY_JAEGER="${DEPLOY_JAEGER:-false}"
+ENABLE_FLUENTBIT="${ENABLE_FLUENTBIT:-false}"
+DEPLOY_PROMETHEUS="${DEPLOY_PROMETHEUS:-false}"
+DEPLOY_WITH_SAME_SA="${DEPLOY_WITH_SAME_SA:-false}"
+ENVOY_LOG_LEVEL="${ENVOY_LOG_LEVEL:-debug}"
+DEPLOY_ON_OPENSHIFT="${DEPLOY_ON_OPENSHIFT:-false}"
+MULTICLUSTER_CONTEXTS="${MULTICLUSTER_CONTEXTS:-()}"
+
+# For any additional installation arguments. Used heavily in CI.
+optionalInstallArgs=$*
+
+exit_error() {
+    error="$1"
+    echo "$error"
+    exit 1
+}
+
+# Check if Docker daemon is running
+docker info > /dev/null || { echo "Docker daemon is not running"; exit 1; }
+
+# Build OSM binaries
+make build-osm
+
+echo "Kubernetes contexts to be deployed to: $MULTICLUSTER_CONTEXTS"
+
+for CONTEXT in $MULTICLUSTER_CONTEXTS; do
+    # shellcheck disable=SC2034
+    BOOKSTORE_CLUSTER_ID="$CONTEXT" # this is used when deploying bookstore app
+
+    kubectl config use-context "$CONTEXT"
+
+    # cleanup stale resources from previous runs
+    ./demo/clean-kubernetes.sh
+
+    # The demo uses osm's namespace as defined by environment variables, K8S_NAMESPACE
+    # to house the control plane components.
+    #
+    # Note: `osm install` creates the namespace via Helm only if such a namespace already
+    # doesn't exist. We explicitly create the namespace below because of the need to
+    # create container registry credentials in this namespace for the purpose of testing.
+    # The side effect of creating the namespace here instead of letting Helm create it is
+    # that Helm no longer manages namespace creation, and as a result labels that it
+    # otherwise adds for using as a namespace selector are no longer available.
+    kubectl create namespace "$K8S_NAMESPACE"
+    # Mimic Helm namespace label behavior: https://github.com/helm/helm/blob/release-3.2/pkg/action/install.go#L292
+    kubectl label namespace "$K8S_NAMESPACE" name="$K8S_NAMESPACE"
+
+    echo "Certificate Manager in use: $CERT_MANAGER"
+    if [ "$CERT_MANAGER" = "vault" ]; then
+        echo "Installing Hashi Vault"
+        ./demo/deploy-vault.sh
+    fi
+
+    if [ "$CERT_MANAGER" = "cert-manager" ]; then
+        echo "Installing cert-manager"
+        ./demo/deploy-cert-manager.sh
+    fi
+
+    if [ "$DEPLOY_ON_OPENSHIFT" = true ] ; then
+        optionalInstallArgs+=" --set=OpenServiceMesh.enablePrivilegedInitContainer=true"
+    fi
+
+    # Push to registry - needs to happen after registry creation
+    make docker-push
+
+    # Registry credentials
+    ./scripts/create-container-registry-creds.sh "$K8S_NAMESPACE"
+
+
+    # Deploys Xds and Prometheus
+    echo "Certificate Manager in use: $CERT_MANAGER"
+    if [ "$CERT_MANAGER" = "vault" ]; then
```

Comment:
We could remove this block. We would not need optionality for the purposes of the demo, since the demo is supposed to show whether/how multicluster OSM works. (One would assume if this works w/ one CA it will work with another.)